### PR TITLE
feat: use aks windows vhds by default

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:
Use aks VHDs as default images for Windows nodes

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/defaults.go [hunk 1]
```diff
@@ -701,11 +701,11 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool, cloudName
 func (p *Properties) setWindowsProfileDefaults(isUpgrade, isScale bool) {
 	windowsProfile := p.WindowsProfile
 	if !isUpgrade && !isScale {
-		if windowsProfile.WindowsPublisher == "" {
-			windowsProfile.WindowsPublisher = DefaultWindowsPublisher
-		}
 
 		if p.IsAzureStackCloud() {
+			if windowsProfile.WindowsPublisher == "" {
+				windowsProfile.WindowsPublisher = DefaultWindowsPublisher
```
Comment (PatrickLang) [inline pkg/api/defaults.go]: @rjaini - do you want to change the AzureStack default in a separate PR or this one?

Comment (rjaini) [reply pkg/api/defaults.go]: yes. As the VHD issue is not resolve yet.

File: pkg/api/const.go [hunk 2]
```diff
@@ -288,6 +288,14 @@ const (
 	DefaultImageVersion = "17763.737.1909062324"
 )
 
+// Aks windows VHD parts
+const (
```
Comment (jackfrancis) [inline pkg/api/const.go]: Is there any practical way to converge where we maintain Windows VHD default references and where we maintain the Linux defaults? See here for Linux VHD config:

https://github.com/Azure/aks-engine/blob/master/pkg/api/azenvtypes.go#L104

Comment (marosset) [reply pkg/api/const.go]: Hmm....  we could add now image config types there and then just grab the values we need installed of looking them up in a 'disto map'.
That wouldn't change the behavior in code but would keep all the image related changes to a single file.
Let me give that a try.

Comment (jackfrancis) [reply pkg/api/const.go]: Yeah I assume things are functionally ready here, just observing that we shouldn't be triaging Windows from Linux coding interfaces if we can help it. (A good justification for doing so would be that the existing Linux implementation has lots of tech debt and the Windows surface area is an excuse to improve maintenance. :))

Comment (marosset) [reply pkg/api/const.go]: Great sell!

Comment (marosset) [reply pkg/api/const.go]: @jackfrancis can you take a look at the lastest commit? I feel it is too big of a change to use how windows images are refenced internal to aks-engine but i did move the defaults align how defaults are specified into azenvtypes.go to align with how linux images are specified.

File: pkg/api/azenvtypes.go [hunk 3]
```diff
@@ -148,6 +148,30 @@ var (
...
+	// AKSWindows2019OSImageConfig is the AKS image based on Windows Server 2019
+	AKSWindowsServer2019OSImageConfig = AzureOSImageConfig{
+		ImageOffer: "aks-windows",
+		ImageSku: "2019-datacenter-core-smalldisk",
+		ImagePublisher: "microsoft-aks",
+		ImageVersion: "17763.737.190923",
+	}
+
+	// AzureStackWindowsServer2019OSImageConfig is the 'vanilla' Windows Server 2019 image to be used to with Azure Stack deployments
+	AzureStackWindowsServer2019OSImageConfig = AzureOSImageConfig{
```
Comment (marosset) [inline pkg/api/azenvtypes.go]: @rjaini as FYI - this change moves where the default imaging values for AzureStack gets defined (from consts.go to azenvtypes.go)

Comment (jackfrancis) [reply pkg/api/azenvtypes.go]: Thanks! I think this fits in better with the existing implementation. (Hope you agree and aren't under the delusion that folks just tell you what to do :/)

Comment (marosset) [reply pkg/api/azenvtypes.go]: I'm unhappy with how different image specification works for Linux vs Windows. 
I looked into unifying it but it looked like it was going to be very painful to do so and support existing API models. 
I would like to see converge for V2 tho.

**Pull request merged**